File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] result = new int[score.length];
        
        List<Integer> list = new ArrayList<>();
        int min = 0;
        
        for(int i = 0; i<score.length; i++){
            
            if(list.size() < k){
                list.add(score[i]);
            } 
            
            else{
                if(min < score[i]){
                    list.remove(Integer.valueOf(min));
                    list.add(score[i]);         
                }
                
            }
            
            min = Collections.min(list);
            
            result[i] = min;
            
        
        }
        return result;
    }
}

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    honer_stage = []
    for i in score:     
        honer_stage.append(i)
        if len(honer_stage) > k:
            honer_stage.remove(min(honer_stage))
        answer.append(min(honer_stage))
        
    
    return answer